OrderService/OrderService.Application/UseCases/OrderUseCases/CreateByBasket/CreateOrdersByBasketRequestHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿using AutoMapper;
+using MediatR;
+using OrderService.Application.Common.Dtos;
+using OrderService.Application.Common.Intefaces;
+using OrderService.Domain.Enums;
+using OrderService.Domain.Exceptions;
+using OrderService.Domain.Interfaces;
+using OrderService.Domain.Models;
+
+namespace OrderService.Application.UseCases.OrderUseCases.CreateByBasket
+{
+    internal sealed class CreateOrdersByBasketRequestHandler(
+        IBasketRepository basketRepository,
+        IProductService productService,
+        IOrderRepository orderRepository,
+        IUnitOfWork unitOfWork,
+        IMapper mapper) : IRequestHandler<CreateOrdersByBasketRequest>
+    {
+        public async Task Handle(CreateOrdersByBasketRequest request, CancellationToken cancellationToken)
+        {
+            var basket = await basketRepository.GetByUserIdWithTrackingAsync(request.UserId, cancellationToken)
+                ?? throw new NotFoundException("There is no basket with such userId");
+
+            if (!basket.BasketItems.Any())
+            {
+                throw new BadRequestException("There are no items in the basket");
+            }
+
+            var productsToTake = mapper.Map<List<TakeProductDto>>(basket.BasketItems);
+
+            var takedProducts = productService.TakeProducts(productsToTake);
+
+            foreach(var product in takedProducts) 
+            {
+                var orderId = Guid.NewGuid();
+                var totalPrice = product.Price * product.Quantity;
+
+                var order = new Order
+                {
+                    Id = orderId,
+                    BuyerId = request.UserId,
+                    SellerId = product.UserId,
+                    Quantity = product.Quantity,
+                    ToTalPrice = totalPrice,
+                    Status = OrderStatus.Processing.ToString(),
+                    OrderItem = new OrderItem
+                    {
+                        Id = Guid.NewGuid(),
+                        OrderId = orderId,
+                        ProductId = product.Id,
+                        ProdcutName = product.Name,
+                        ProductPrice = product.Price
+                    }
+                };
```

Comment:
how user can create many orders at the same time?

Reply:
Each product has its own manufacturer and that is why separate orders are created.

Reply:
ok, then please move creation of the order to a separate method